src/cargo/core/compiler/job_queue.rs

```diff
@@ -350,6 +351,7 @@ impl<'a, 'cfg> JobQueue<'a, 'cfg> {
                         print.print(&msg)?;
                     }
                     Message::Finish(id, artifact, result) => {
+                        finished += 1;
```

Comment:
I think this needs to be moved down to the `Artifact::All` branch?

Reply:
Hm yes indeed